File: packages/sdk/go/internal/model/audio.go
// Package model 的音频(TTS / ASR / 语音对话 / 声纹 / 音乐)类型。
package model

// AudioSpeechRequest 是 POST /v1/audio/speech 请求体(TTS)。
type AudioSpeechRequest struct {
	Model          string  `json:"model"`
	Input          string  `json:"input"`
	Voice          string  `json:"voice"`
	ResponseFormat string  `json:"responseFormat,omitempty"`
	Speed          float64 `json:"speed,omitempty"`
}

// AudioSpeechResponse 是 POST /v1/audio/speech 响应体(二进制音频流,base64 编码)。
type AudioSpeechResponse struct {
	Audio      string `json:"audio"`
	Format     string `json:"format"`
	DurationMs int64  `json:"durationMs"`
}

// AudioTranscriptionsRequest 是 POST /v1/audio/transcriptions 请求体(ASR)。
type AudioTranscriptionsRequest struct {
	Model    string `json:"model"`
	Audio    string `json:"audio"`
	Language string `json:"language,omitempty"`
	Prompt   string `json:"prompt,omitempty"`
}

// AudioTranscriptionSegment 是 ASR 转录片段。
type AudioTranscriptionSegment struct {
	ID    int     `json:"id"`
	Start float64 `json:"start"`
	End   float64 `json:"end"`
	Text  string  `json:"text"`
}

// AudioTranscriptionsResponse 是 POST /v1/audio/transcriptions 响应体。
type AudioTranscriptionsResponse struct {
	Text     string                      `json:"text"`
	Language string                      `json:"language"`
	Duration float64                     `json:"duration"`
	Segments []AudioTranscriptionSegment `json:"segments,omitempty"`
}

// AudioVoiceItem 是音色列表项。
type AudioVoiceItem struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Gender   string `json:"gender"`
	Language string `json:"language"`
	Preview  string `json:"preview,omitempty"`
}

// AudioVoicesResponse 是 GET /v1/audio/voices 响应体。
type AudioVoicesResponse struct {
	Object string           `json:"object"`
	Data   []AudioVoiceItem `json:"data"`
}

// AudioChatRequest 是 POST /v1/audio/chat 请求体(语音对话)。
type AudioChatRequest struct {
	Audio     string `json:"audio"`
	Model     string `json:"model"`
	SessionID string `json:"sessionId,omitempty"`
}

// AudioChatResponse 是 POST /v1/audio/chat 响应体。
type AudioChatResponse struct {
	Text      string `json:"text"`
	Audio     string `json:"audio"`
	SessionID string `json:"sessionId"`
}

// RegisterSpeakerRequest 是 POST /v1/audio/speakers 请求体(声纹注册)。
type RegisterSpeakerRequest struct {
	Name  string `json:"name"`
	Audio string `json:"audio"`
}

// RegisterSpeakerResponse 是声纹注册响应。
type RegisterSpeakerResponse struct {
	SpeakerID string `json:"speakerId"`
	Status    string `json:"status"`
}

// SpeakersListResponse 是 GET /v1/audio/speakers 响应体(声纹列表)。
type SpeakersListResponse struct {
	Object string `json:"object"`
	Data   []struct {
		ID           string `json:"id"`
		Name         string `json:"name"`
		RegisteredAt string `json:"registeredAt"`
	} `json:"data"`
}

// CompareSpeakersRequest 是 POST /v1/audio/speakers/compare 请求体(声纹比对)。
type CompareSpeakersRequest struct {
	SpeakerID string `json:"speakerId"`
	Audio     string `json:"audio"`
}

// CompareSpeakersResponse 是 POST /v1/audio/speakers/compare 响应体。
type CompareSpeakersResponse struct {
	Score   float64 `json:"score"`
	Matched bool    `json:"matched"`
}

// MusicGenerationsRequest 是 POST /v1/audio/music 请求体(音乐生成)。
type MusicGenerationsRequest struct {
	Prompt   string `json:"prompt"`
	Lyrics   string `json:"lyrics,omitempty"`
	Duration *int   `json:"duration,omitempty"`
}

// MusicGenerationsResponse 是 POST /v1/audio/music 响应体。
type MusicGenerationsResponse struct {
	TaskID string `json:"taskId"`
	Status string `json:"status"`
}

File: sdks/typescript/src/audio.ts
/**
 * 音频模块 — TTS / ASR / 语音对话 / 声纹 / 音乐生成。
 *
 * 端点(8 个):
 * - GET  /v1/audio/voices
 * - POST /v1/audio/speech(TTS)
 * - POST /v1/audio/transcriptions(ASR)
 * - POST /v1/audio/chat(语音对话)
 * - GET  /v1/audio/speakers(声纹列表)
 * - POST /v1/audio/speakers(声纹注册)
 * - POST /v1/audio/speakers/compare(声纹比对)
 * - POST /v1/audio/music(音乐生成)
 */

import type { BaseClient } from './base.js'

// =============================================================================
// 内联类型定义
// =============================================================================

export interface V1AudioVoicesResponse {
  object: 'list'
  data: Array<{
    id: string
    name: string
    gender: string
    language: string
    preview?: string
  }>
}

export interface V1AudioSpeechRequest {
  model: string
  input: string
  voice: string
  responseFormat?: string
  speed?: number
}

export interface V1AudioSpeechResponse {
  audio: string
  format: string
  durationMs: number
}

export interface V1AudioTranscriptionsRequest {
  model: string
  audio: string
  language?: string
  prompt?: string
}

export interface V1AudioTranscriptionsResponse {
  text: string
  language: string
  duration: number
  segments?: Array<{
    id: number
    start: number
    end: number
    text: string
  }>
}

export interface V1AudioChatRequest {
  audio: string
  model: string
  sessionId?: string
}

export interface V1AudioChatResponse {
  text: string
  audio: string
  sessionId: string
}

export interface V1RegisterSpeakerRequest {
  name: string
  audio: string
}

export interface V1CompareSpeakersRequest {
  speakerId: string
  audio: string
}

export interface V1CompareSpeakersResponse {
  score: number
  matched: boolean
}

export interface V1MusicGenerationsRequest {
  prompt: string
  lyrics?: string
  duration?: number
}

export interface V1MusicGenerationsResponse {
  taskId: string
  status: 'pending' | 'processing' | 'completed'
}

/** 声纹注册响应。 */
export interface V1RegisterSpeakerResponse {
  speakerId: string
  status: 'registered'
}

/** 声纹列表响应。 */
export interface V1SpeakersListResponse {
  object: 'list'
  data: Array<{ id: string; name: string; registeredAt: string }>
}

// =============================================================================
// Module 接口 + 工厂函数
// =============================================================================

export interface AudioModule {
  /** GET /v1/audio/voices(音色列表)。 */
  listVoices(): Promise<V1AudioVoicesResponse>
  /** POST /v1/audio/speech(文字转语音)。 */
  speech(req: V1AudioSpeechRequest): Promise<V1AudioSpeechResponse>
  /** POST /v1/audio/transcriptions(语音转文字)。 */
  transcriptions(req: V1AudioTranscriptionsRequest): Promise<V1AudioTranscriptionsResponse>
  /** POST /v1/audio/chat(语音对话)。 */
  chat(req: V1AudioChatRequest): Promise<V1AudioChatResponse>
  /** GET /v1/audio/speakers(声纹列表)。 */
  listSpeakers(): Promise<V1SpeakersListResponse>
  /** POST /v1/audio/speakers(声纹注册)。 */
  registerSpeaker(req: V1RegisterSpeakerRequest): Promise<V1RegisterSpeakerResponse>
  /** POST /v1/audio/speakers/compare(声纹比对)。 */
  compareSpeakers(req: V1CompareSpeakersRequest): Promise<V1CompareSpeakersResponse>
  /** POST /v1/audio/music(音乐生成)。 */
  music(req: V1MusicGenerationsRequest): Promise<V1MusicGenerationsResponse>
}

export function createAudioModule(client: BaseClient): AudioModule {
  return {
    listVoices: () => client.request<V1AudioVoicesResponse>('GET', '/audio/voices'),
    speech: (req) => client.request<V1AudioSpeechResponse>('POST', '/audio/speech', req),
    transcriptions: (req) =>
      client.request<V1AudioTranscriptionsResponse>('POST', '/audio/transcriptions', req),
    chat: (req) => client.request<V1AudioChatResponse>('POST', '/audio/chat', req),
    listSpeakers: () => client.request<V1SpeakersListResponse>('GET', '/audio/speakers'),
    registerSpeaker: (req) =>
      client.request<V1RegisterSpeakerResponse>('POST', '/audio/speakers', req),
    compareSpeakers: (req) =>
      client.request<V1CompareSpeakersResponse>('POST', '/audio/speakers/compare', req),
    music: (req) => client.request<V1MusicGenerationsResponse>('POST', '/audio/music', req),
  }
}
